pkg/webhook/resources/virtualmachine/mutator.go

```diff
@@ -534,6 +534,44 @@ func (m *vmMutator) patchTerminationGracePeriodSeconds(vm *kubevirtv1.VirtualMac
 	return patchOps, nil
 }
 
+func patchCPUSocket(vm *kubevirtv1.VirtualMachine, patchOps types.PatchOps) []string {
+	cores := vm.Spec.Template.Spec.Domain.CPU.Cores
+	sockets := vm.Spec.Template.Spec.Domain.CPU.Sockets
+	threads := vm.Spec.Template.Spec.Domain.CPU.Threads
+
+	if cores == 0 {
+		cores = 1
+	}
+	if sockets == 0 {
+		sockets = 1
+	}
+	if threads == 0 {
+		threads = 1
+	}
+
+	patchSocket := false
+	if cores != 1 {
```

Comment:
if original cores == "2", here it is also replaced to "1", is it expected? similar question on others

or:
```
if vm.Spec.Template.Spec.Domain.CPU.Cores == 0 {
    patchOps = append
}
```

Reply:
This is expected, because we want to use `sockets` to represent all CPU.

Reply:
I thought the only place where the CPU is mapped to the socket is the UI. If user wants to change the CPU topology via YAML, I don't see any reasons to disallow that. Or if we really don't want user to mess with cores and threads, then we can consider using the validator to reject the request. I think the validator is easier to reason about compared with trying to reset the user-provided values to 1.

Reply:
There is `DedicatedCPUPlacement` case, if the cores is replaced to `1` all the time, will it break CPU pinning?  In kubevirt code, there is also processing related to CPU pinning. cc @brandboat, please also double check, thanks.

```
func isDedicatedCPU(vm *kubevirtv1.VirtualMachine) bool {
	return vm.Spec.Template.Spec.Domain.CPU != nil && vm.Spec.Template.Spec.Domain.CPU.DedicatedCPUPlacement
}
```

Reply:
> There is DedicatedCPUPlacement case, if the cores is replaced to 1 all the time, will it break CPU pinning? In kubevirt code, there is also processing related to CPU pinning. cc @brandboat, please also double check, thanks.

No it won't, I just test the VM with the following settings
```
      domain:
        cpu:
          cores: 1
          dedicatedCpuPlacement: true
          sockets: 2
          threads: 1
```
everything is ok.
```
qemu@t:/> virsh dumpxml default_t | awk "/<cputune>/,/<\/cputune>/"
  <cputune>
    <vcpupin vcpu='0' cpuset='1'/>
    <vcpupin vcpu='1' cpuset='2'/>
  </cputune>
```

Reply:
> BTW, In the hot-plug + cpu pin scenario, there is a chance that when expanding vCPUs and there is no target node meeting the required dedicated cpus, then at which stage will it fail? trigger hot-plug, trigger migration, migration is not scheduable? thanks.

I'm not familiar with the hot-plug part, but from what I know, if the target node doesn't have enough dedicated CPUs, the virt-launcher pod will just stay pending until the node has enough resources to continue.